SPARQLWrapper/Wrapper.py

```diff
@@ -1,1369 +1,1373 @@
+            * in the case of RDF :data:`Turtle<TURTLE>`/:data:`N3`, a string is returned
+            * in the case of :data:`CSV`/:data:`TSV`, a string is returned
+            * In all other cases the input simply returned.
+
+        :return: the converted query result. See the conversion methods for more details.
+        """
+
+
+        def _content_type_in_list(real: str, expected: List[str]) -> bool:
+            """Internal method for checking if the content-type header received matches any of the content types of
+            the expected list.
+
+            :param real: The content-type header received.
+            :type real: string
+            :param expected: A list of expected content types.
+            :type expected: list
+            :return: Returns a boolean after checking if the content-type header received matches any of the content
+            types of the expected list.
+            :rtype: boolean
+            """
+            return True in [real.find(mime) != -1 for mime in expected]
+
+        def _validate_format(
+            format_name: str, allowed: List[str], mime: str, requested: str
+        ) -> None:
+            """Internal method for validating if the requested format is one of the allowed formats.
+
+            :param format_name: The format name (to be used in the warning message).
+            :type format_name: string
+            :param allowed: A list of allowed content types.
+            :type allowed: list
+            :param mime: The content-type header received (to be used in the warning message).
+            :type mime: string
+            :param requested: the requested format.
+            :type requested: string
+            """
+            if requested not in allowed:
+                message = "Format requested was %s, but %s (%s) has been returned by the endpoint"
+                warnings.warn(
+                    message % (requested.upper(), format_name, mime), RuntimeWarning
+                )
+        # TODO. In order to compare properly, the requested QueryType (SPARQL Query Form) is needed. For instance,
+        # the unexpected N3 requested for a SELECT would return XML
+        if "content-type" in self.info():
+            ct = self.info()["content-type"]  # returned Content-Type value
+            
+            if _content_type_in_list(ct, _SPARQL_XML):
+                _validate_format("XML", [XML], ct, self.requestedFormat)
+                return self._convertXML()
+            elif _content_type_in_list(ct, _XML):
+                _validate_format("XML", [XML], ct, self.requestedFormat)
+                return self._convertXML()
+            elif _content_type_in_list(ct, _SPARQL_JSON):
+                _validate_format("JSON", [JSON], ct, self.requestedFormat)
+                return self._convertJSON()
+            elif _content_type_in_list(ct, _RDF_XML):
+                #Ananya: This has to be altered

```

Comment:
Please remove this.

Reply:
Removing these would cause an error some asserts in the test file.
Altering those too.